File: library/src/main/java/cz/ackee/resizin/Gravity.java
package cz.ackee.resizin;

/**
 * Gravity of {@link Crop}
 */
public enum Gravity {
    S("s"),
    N("n"),
    W("w"),
    E("e"),
    SW("sw"),
    SE("se"),
    NW("nw"),
    NE("ne"),
    CENTER("center"),
    FACE("face");

    private final String value;

    Gravity(String value) {
        this.value = value;
    }

    public String toString() {
        return this.value;
    }


    public static Gravity fromValue(String value) {
        for (Gravity c : values()) {
            if (c.value.equals(value)) {
                return c;
            }
        }
        throw new IllegalArgumentException("Invalid value " + value);
    }
}

File: sdk/src/main/java/cz/ackee/resizin/Gravity.kt
package cz.ackee.resizin

/**
 * Gravity of [Crop]
 */
enum class Gravity private constructor(private val value: String) {

    S("s"),
    N("n"),
    W("w"),
    E("e"),
    SW("sw"),
    SE("se"),
    NW("nw"),
    NE("ne"),
    CENTER("center"),
    FACE("face");

    override fun toString() = this.value

    companion object {

        fun fromValue(value: String): Gravity {
            return values().firstOrNull { it.value == value } ?: throw IllegalArgumentException("Invalid value $value")
        }
    }
}
